### ISSUE TITLE
Generating customized talos boot assets doesn't add any system extension assets

### ISSUE THREAD

**Issue by trunet** [siderolabs/talos#7924]
## Bug Report

I'm trying to generate a custom boot asset (iso, initramfs, whatever), and I can't manage to have zfs or any other extension assets binaries into it.

### Description

Bottom line, I'm trying to have zfs extension added to talos. I'm some days in it and I ran out of ideas.

My first try was the "deprecated" machine.installer. I added zfs there (together with other extensions). It adds the kernel modules, but there're no binaries anywhere in the talos filesystem (like /usr/local/sbin/zfs, or any other bin places). If I spin an alpine pod, bind mount /dev with the host /dev and install zfs, it works, can create pools, so on. But there's no zfs binaries in the host/node filesystem, only the kernel modules were added. This is already odd, I guess it should work and it's not, but not what this issue is about, moving on...

Second try, using https://www.talos.dev/v1.5/talos-guides/install/boot-assets/ installation guide:

```
❯ docker run --rm -t -v $PWD/_out:/out ghcr.io/siderolabs/imager:v1.5.4 iso \
  --system-extension-image ghcr.io/siderolabs/qemu-guest-agent:8.0.2@sha256:79979c41a8acfc51b2651d6d44bf187b8e7164e6c649f4e39b9cb699fed8917b \
  --system-extension-image ghcr.io/siderolabs/iscsi-tools:v0.1.4@sha256:58cbadb0a315d83e04b240de72c5ac584e9c63b690b2c4fcd629dd1566c3a7a1 \
  --system-extension-image ghcr.io/siderolabs/zfs:2.1.12-v1.5.4@sha256:df5674ed21a97c96ec4903ebdbd7318bb8ab5d18fbf10070139f84bf72a94d2d \
  --extra-kernel-arg net.ifnames=0
profile ready:
arch: amd64
platform: metal
secureboot: false
version: v1.5.4
customization:
  extraKernelArgs:
    - net.ifnames=0
input:
  kernel:
    path: /usr/install/amd64/vmlinuz
  initramfs:
    path: /usr/install/amd64/initramfs.xz
  baseInstaller:
    imageRef: ghcr.io/siderolabs/installer:v1.5.4
  systemExtensions:
    - imageRef: ghcr.io/siderolabs/qemu-guest-agent:8.0.2@sha256:79979c41a8acfc51b2651d6d44bf187b8e7164e6c649f4e39b9cb699fed8917b
    - imageRef: ghcr.io/siderolabs/iscsi-tools:v0.1.4@sha256:58cbadb0a315d83e04b240de72c5ac584e9c63b690b2c4fcd629dd1566c3a7a1
    - imageRef: ghcr.io/siderolabs/zfs:2.1.12-v1.5.4@sha256:df5674ed21a97c96ec4903ebdbd7318bb8ab5d18fbf10070139f84bf72a94d2d
output:
  kind: iso
  outFormat: raw
initramfs ready
kernel command line: talos.platform=metal console=ttyS0 console=tty0 init_on_alloc=1 slab_nomerge pti=on consoleblank=0 nvme_core.io_timeout=4294967295 printk.devkmsg=on ima_template=ima-ng ima_appraise=fix ima_hash=sha512 net.ifnames=0
ISO ready
output asset path: /out/metal-amd64.iso
```

Now, if you mount the ISO, and `unsquashfs rootfs.sqsh`, there's no iscsi tools, no zfs, no extension stuff added to it.

Third try, using https://factory.talos.dev/. Selected the extensions, downloaded the generated ISO, and same thing as my local one there's no extension stuff in it. My schematic ID is `89fd8ccd0a79823312879d0758aa3f1a47c1b7b364e8979a60f87d9737bc1171`.

**Comment by smira** [comment]
There are no system extensions in `rootfs.sqsh`, so the issue is irrelevant. The extensions are part of `initramfs.xz`, and hard to decode with CLI tools. 

I guess you're not using proper `installer` with extensions, so the extensions are part of the initial boot, but not part of your install.

The flow with Image Factory is documented in https://www.talos.dev/v1.5/talos-guides/install/boot-assets/#example-bare-metal-with-image-factory

If I download the ISO with your schematic and boot from it:

```
$ curl -LO https://factory.talos.dev/image/89fd8ccd0a79823312879d0758aa3f1a47c1b7b364e8979a60f87d9737bc1171/v1.5.4/metal-amd64.iso
$ talosctl -n 172.20.0.2 get extensions -i
NODE   NAMESPACE   TYPE              ID            VERSION   NAME               VERSION
       runtime     ExtensionStatus   0             1         zfs                2.1.12-v1.5.4
       runtime     ExtensionStatus   1             1         iscsi-tools        v0.1.4
       runtime     ExtensionStatus   2             1         qemu-guest-agent   8.0.2
       runtime     ExtensionStatus   3             1         schematic          89fd8ccd0a79823312879d0758aa3f1a47c1b7b364e8979a60f87d9737bc1171
       runtime     ExtensionStatus   modules.dep   1         modules.dep        6.1.58-talos
```

The extensions are installed as expected. Please follow the documentation and ensure you're using proper `installer` image.

**Comment by trunet** [comment]
Thanks for checking... It's not clear to me how the extensions are actually installed and added to the talos filesystem. Looks like the zfs extension for talos 1.5.4, doesn't have the ZFS binaries. https://github.com/siderolabs/extensions/issues/255 says that it works if I jump to 1.6 alpha.

**Comment by frezbo** [comment]
> Thanks for checking... It's not clear to me how the extensions are actually installed and added to the talos filesystem. Looks like the zfs extension for talos 1.5.4, doesn't have the ZFS binaries. [siderolabs/extensions#255](https://github.com/siderolabs/extensions/issues/255) says that it works if I jump to 1.6 alpha.

zfs tools would be only available from 1.6, we don't backport new additions, only bugs/security fixes
